docs/cores.rst

```diff
@@ -713,27 +848,78 @@ Options
 
 Set ``rest_detailed_fields`` if you want to define fields that will be returned inside REST response for a request on
 concrete object (an URL contains an ID of a concrete model object. For example an URL of a request is ``/api/user/1/``).
-This option rewrites settings inside ``RESTMeta`` (you can find more about it at section #TODO add link).
 
 .. attribute:: DjangoRestCore.rest_general_fields
 
 Set ``rest_general_fields`` if you want to define fields that will be returned inside REST response for a request on
 more than one object (an URL does not contain an ID of a concrete objects, eq. ``/api/user/``). This defined set of
-fields is used for generating result of a foreign key object. This option rewrites settings inside ``RESTMeta``
-(you can find more about it at section #TODO add link).
+fields is used for generating result of a foreign key object.
 
 .. attribute:: DjangoRestCore.rest_extra_fields
 
 Use ``rest_extra_fields`` to define extra fields that is not returned by default, but can be extra requested
-by a HTTP header ``X-Fields`` or a GET parameter ``_fields``. More info you can find in **django-piston** library
-documentation. This option rewrites settings inside ``RESTMeta`` (you can find more about it at section #TODO add link).
+by a HTTP header ``X-Fields`` or a GET parameter ``_fields``. See the django-pyston library documentation for more details.
+
+**Example: API-Only Fields**
+
+Expose computed or aggregated data via REST API without including it in default responses.
+
+Define the fields in your Core::
+
+    class CustomerCore(DjangoUiRestCore):
+        model = Customer
+
+        # Fields only available when explicitly requested via API
+        rest_extra_fields = (
+            'coin_balance',           # Computed field
+            'applied_promocodes_id',  # Filterable field
+        )
+
+        # Allow filtering by extra fields in REST API
+        rest_extra_filter_fields = (
+            'applied_promocodes_id',
+        )
+
+Then define the computed fields in your Resource::
+
+    from is_core.utils.decorators import short_description
+    from pyston.utils.decorators import filter_class
+
+    class CustomerResource(DjangoCoreResource):
+        model = Customer
+
+        @short_description(_('Coin Balance'))
+        def coin_balance(self, obj):
+            return CoinTransaction.objects.compute_balance_for_customer(obj)
+
+        # For filterable extra fields, use @filter_class decorator
+        @filter_class(AppliedPromocodeIDFilter)
+        @property
+        def applied_promocodes_id(self):
+            return None  # Return value not used, filter handles the logic
```

Comment:
not returning any value means `rest_extra_fields` will always return `None` for this property. We should specify that.